xlive/Blam/Engine/input/input_abstraction.h

```diff
@@ -1,3 +1,217 @@
 #pragma once
 
+#include "controllers.h"
+#include "input_windows.h"
+
+#define K_MAXIMUM_NUMBER_OF_GAME_FUNCTION_BINDS 8
+
+enum e_input_preference_device_type :uint32
+{
+	_input_preference_device_general = 0xFFFFFFFF,
+	_input_preference_device_mouse = 0x0,
+	_input_preference_device_keyboard = 0x1,
+	_input_preference_device_gamepad = 0x2,
+};
+
+enum e_button_functions
+{
+	_button_jump = 0x0,
+	_button_trick = 0x1,
+	_button_brake = 0x2,
+	_button_switch_grenade = 0x3,
+	_button_switch_weapon = 0x4,
+	_button_melee_attack = 0x5,
+	_button_flashlight = 0x6,
+	_button_throw_grenade = 0x7,
+	_button_speed_boost = 0x8,
+	_button_e_brake = 0x9,
+	_button_fire = 0xA,
+	_button_start = 0xB,
+	_button_back = 0xC,
+	_button_crouch = 0xD,
+	_button_scope_zoom = 0xE,
+	_button_scope_zoom_in = 0xF,
+	_button_scope_zoom_out = 0x10,
+	_button_lean_left = 0x11,
+	_button_lean_right = 0x12,
+	_button_accept = 0x13,
+	_button_cancel = 0x14,
+	_button_banshee_bomb = 0x15,
+	_button_dual_wield_primary_fire = 0x16,
+	_button_dual_wield_secondary_fire = 0x17,
+	_button_vehicle_primary_fire = 0x18,
+	_button_vehicle_secondary_fire = 0x19,
+	_button_text_chat_toggle = 0x1A,
+	_button_text_chat_team = 0x1B,
+	_button_ui_scroll_up = 0x1C,
+	_button_ui_scroll_down = 0x1D,
+	_button_team_voice = 0x1E,
+	_button_flip_vehicle = 0x1F,
+	_button_touch_device = 0x20,
+	_button_enter_vehicle = 0x21,
+	_button_board_vehicle = 0x22,
+	_button_evict_from_vehicle = 0x23,
+	_button_trade_weapon = 0x24,
+	_button_pick_up_primary_weapon = 0x25,
+	_button_pick_up_primary_multiplayer_weapon = 0x26,
+	_button_pick_up_secondary_weapon = 0x27,
+	_button_pick_up_secondary_multiplayer_weapon = 0x28,
+	_button_put_away_secondary_weapon = 0x29,
+	_button_put_away_or_drop_secondary_weapon = 0x2A,
+	_button_reload = 0x2B,
+	_button_exit_vehicle = 0x2C,
+	_button_move_forward = 0x2D,
+	_button_move_backward = 0x2E,
+	_button_strafe_left = 0x2F,
+	_button_strafe_right = 0x30,
+	_button_mouse_yaw_left = 0x31,
+	_button_mouse_yaw_right = 0x32,
+	_button_mouse_pitch_forward = 0x33,
+	_button_mouse_pitch_backward = 0x34,
+	_extended_button_gamepad_yaw_left = 0x35,
+	_extended_button_gamepad_yaw_right = 0x36,
+	_extended_button_gamepad_pitch_forward = 0x37,
+	_extended_button_gamepad_pitch_backward = 0x38,
+
+	NUMBER_OF_BUTTON_KEYS = 0x39,
+	NUMBER_OF_ABSTRACT_BUTTONS = 0x2D,
```

Comment:
rename these to use the k_ convention

Reply:
its actually called NUMBER_OF_EXTENDED_CONTROL_BUTTONS